force-app/main/default/classes/AttendanceService.cls

```diff
@@ -42,6 +44,75 @@ public with sharing class AttendanceService {
         }
     }
 
+    public Map<String, Boolean> checkFieldPermissions() {
+        Map<String, Boolean> result = new Map<String, Boolean>();
```

Comment:
What do you think about adding a method to PermissionValidator that takes a list of SObjectFields, we are converting to string then back to describe. It might be a little easier to read if we just pass a list of fields?